pilopt/src/lib.rs

```diff
@@ -215,6 +217,44 @@ fn constant_value(function: &FunctionValueDefinition) -> Option<BigUint> {
     }
 }
 
+/// Deduplicate fixed columns of the same namespace which share the same value.
+/// This compares the function values, so `|i| i` is different from `|j| j`
```

Comment:
We could actually quite easily support `|i| i` being equal to `|j| j`, because all references are already the same. The only thing we need to change is the comparison and hash function for `Pattern` to ignore the variable name.

Reply:
Oh great, I was hoping for something like this

Reply:
So I'm looking into this, I don't think we should make all `Pattern::Variable` equal, right?
One level up I was looking at
```rust
#[derive(
    Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, JsonSchema, Hash,
)]
pub struct LambdaExpression<E = Expression<NamespacedPolynomialReference>> {
    pub kind: FunctionKind,
    pub params: Vec<Pattern>,
    pub body: Box<E>,
    /// Type of the parameters, filled in during type inference.
    pub param_types: Vec<Type>,
}
```
Here I would expect `body` to refer to the parameters by index, in which case it would be the same no matter what the parameters are called. Then we could ignore them in comparisons.
Unless I'm missing obvious, this seems like it should be a different PR?

Reply:
I mean the question is if we want to do that in the "default" comparison function. Is there an easy way to provide a different comparison - probably quite cumbersome to do that recursively, right?

Reply:
Opened https://github.com/powdr-labs/powdr/issues/2128 for this